Added Pet model

Flartman opened the pull request:
- new Pet model
- new PetStatus enum
- new Property model

Flartman committed:
Added Pet model

- new Pet model
- new PetStatus enum
- new Property model

YADARAN1 commented on backend/src/GetAPet.Domain/Pet/Pet.cs:
```diff
@@ -0,0 +1,43 @@
+
+        public PetStatus Status { get; }
+
+        public List<Property> Properties { get; } = [];
```
Под проперти может подойти что угодно, переименуй в понятное название

Flartman committed:
Fixed name of payment details class

- Property class was renamed to "PaymentDetails"

- List Properties was renamed to "PaymentDetailsList"
